pkg/endpoint/bpf.go

```diff
@@ -638,9 +639,14 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (revnum uint
 		start := time.Now()
 		// Compile and install BPF programs for this endpoint
 		err = e.runInit(libdir, rundir, epdir, epInfoCache.ifName, debug)
+		duration := time.Since(start)
 		e.getLogger().WithError(err).
-			WithField(logfields.BPFCompilationTime, time.Since(start).String()).
+			WithField(logfields.BPFCompilationTime, duration.String()).
 			Debugf("BPF compilation completed")
+		metrics.EndpointRegenerationTime.With(map[string]string{
+			logfields.LogSubsys:  metrics.LabelEndpointRegenerationBPFCompilation,
+			logfields.EndpointID: e.StringID(),
```

Comment:
The endpointID can be omitted. Prometheus labels should only describe static sets and not temporary or generated labels that have no meaning over time.